core/src/service/sync_service.rs

```diff
@@ -233,23 +242,79 @@ impl RequestContext<'_, '_> {
         Ok(result)
     }
 
-    pub fn get_updates(&self) -> CoreResult<GetUpdatesResponse> {
-        let account = self
+    fn reset_deleted_files(&mut self) -> CoreResult<()> {
+        // resets all changes to files that are implicitly deleted, then explicitly deletes them
+        // we don't want to push updates to deleted documents and we might as well not push updates to deleted metadata
+        // we must explicitly delete a file which is moved into a deleted folder because otherwise resetting it makes it no longer deleted
+        // todo: document repo deletions are not atomic and an interruption during sync can cause digests to become out of date
```

Comment:
formalized/moved logic from merge, handles concerns regarding updates to deleted files